### PR TITLE
Revise libsingular_julia

### PR DESCRIPTION
**Pull request by fingolfin**
The build dependencies on GMP_jll and MPFR_jll were only implicit so far,
but we need to make them explicit and with version constraints to avoid
issues on macOS (when this JLL is built against a too-new version of either
of these libraries, it becomes unloadable in older Julia versions).

Supporting more platforms is kind of an experiment. I am not sure what impediments there are (other than the lack of `libcxxwrap_julia` binaries for some -- but I hope that once libjulia_jll 1.4 is ready, @barche will be willing to update `libcxxwrap_julia` to use `libjulia_jll`, at which point it could probably cover all (?) platforms. But we'll see.

Closes #2203

### CODE REVIEW

**Comment by barche** [comment]
> but I hope that once libjulia_jll 1.4 is ready, @barche will be willing to update `libcxxwrap_julia` to use `libjulia_jll`

Holding out a bit for libjulia 1.3, then I can release a libcxxwrap based on that, immediately followed by one based on 1.4 with a Julia compat requirement of 1.4.

**Comment by fingolfin** [comment]
This is on hold until PR #1948 is resolved.

**Comment by fingolfin** [comment]
@barche @giordano This is now ready for review; see also issue #2160 for an explanation on what's going on. Since this might eventually become template for a bunch of other packages, before merging the whole approach should be scrutinized.

File: L/libsingular_julia/common.jl
Comment (thofma) [inline L/libsingular_julia/common.jl:12]:
```suggestion
              "26da0761f4f8a706f3e3397cc92daf35972edbd9"),
```

File: L/libsingular_julia/common.jl
Comment (fingolfin) [inline L/libsingular_julia/common.jl:7]: Given the new versioning scheme started in #2234 it might make sense to use a variant of that here, too

File: L/libsingular_julia/common.jl
Comment (fingolfin) [inline L/libsingular_julia/common.jl:54]: TODO: add a VersionSpec, something like this (depending on whatever version for Singular_jll we end up with)
```suggestion
    Dependency(PackageSpec(name="Singular_jll", version=VersionSpec("400.100.400-400.100"))),
```

**Comment by fingolfin** [comment]
@thofma @benlorenz I think this is ready -- it changes the version from 0.2.x to 0.3.x, so it should be safe to merge this now, as it won't be "accidentally" used by the current Singular.jl release. Once merged, I'd wait with the corresponding updates for Julia 1.4 and 1.5 until after we confirmed this one works with an updated Singular.jl 

That said, we can't really update Singular.jl to use it until https://github.com/Nemocas/Nemo.jl/pull/919 is merged and Nemo.jl gets a new release...

CC @fieker

**Review by benlorenz — APPROVED** [review]
Looks good, I did a quick test on a gcc9 based linux machine with the binaries from here via override, dev'ing the various downstream packages, adjusting a bunch of project files (versions + deps) and using the flint 2.7 pull request branch:
```
...
   Testing Singular tests passed 
```

**Comment by fingolfin** [comment]
Thanks @benlorenz that's awesome

**Review by thofma — APPROVED** [review]

**Comment by giordano** [comment]
Good to go then?

**Comment by fingolfin** [comment]
Yes, please

**Pull request merged**
